pkgs/by-name/ft/ftb-app/package.nix

```
@@ -0,0 +1,42 @@
+{
+  lib,
+  fetchurl,
+  appimageTools,
+  makeDesktopItem,
+  copyDesktopItems,
+}: let
+  pname = "ftb-app";
+  version = "1.27.2";
+
+  src = fetchurl {
+    url = "https://piston.feed-the-beast.com/app/ftb-app-linux-${version}-x86_64.AppImage";
+    sha256 = "35GEI1OBvVkUvHvQAzzGz8ux9h+5W3acH0Wr5VkqyBw=";
+  };
+in
+  appimageTools.wrapType2 rec {
+    inherit pname src version;
+
+    desktopItem = makeDesktopItem {
+      name = "FTB App";
+      desktopName = "FTB App";
+      exec = "ftb-app";
+      icon = "ftb-app";
+      comment = "Feed the Beast desktop app";
+      categories = ["Game"];
+    };
+
+    extraInstallCommands = ''
+      mkdir -p $out/share/applications
+      cp ${desktopItem}/share/applications/* $out/share/applications/
+    '';
```

Instead of using `desktopItem` and `rec`, do this instead:

```suggestion
  };
  
  appimageContents = appimageTools.extractType2 { inherit pname src version; };
in
  appimageTools.wrapType2 {
    inherit pname src version;
    
    extraInstallCommands = ''
      for size in 16x16 32x32 48x48 64x64 128x128 256x256 512x512; do
        install -Dm644 ${appimageContents}/usr/share/icons/hicolor/$size/apps/ftb-app.png \
          $out/share/icons/hicolor/$size/apps/ftb-app.png
      done
      
      install -Dm644 ${appimageContents}/ftb-app.desktop \
        $out/share/applications/ftb-app.desktop
      substituteInPlace $out/share/applications/ftb-app.desktop \
        --replace-fail 'Exec=AppRun' 'Exec=${placeholder "out"}/bin/ftb-app'
    '';
```

It installs the desktop item from the appimage and also the application icons